src/BBT.StructureTools.Tests/Convert/ConvertRootTreeLeafWithMasterDataTests.cs

```diff
@@ -0,0 +1,118 @@
+﻿namespace BBT.StructureTools.Tests.Convert
+{
+    using System;
+    using System.Collections.Generic;
+    using System.Linq;
+    using BBT.StructureTools.Convert;
+    using BBT.StructureTools.Copy;
+    using BBT.StructureTools.Tests.Convert.Intention;
+    using BBT.StructureTools.Tests.Convert.TestStructure;
+    using BBT.StructureTools.Tests.Convert.TestStructure.Registration;
+    using BBT.StructureTools.Tests.Convert.TestStructure.Source;
+    using BBT.StructureTools.Tests.Convert.TestStructure.Target;
+    using BBT.StructureTools.Tests.TestTools;
+    using FluentAssertions;
+    using Ninject;
+    using Xunit;
+
+    public class ConvertRootTreeLeafWithMasterDataTests
+    {
+        private readonly IConvert<Root, TargetRoot, ITestConvertIntention> testcandidate;
+
+        public ConvertRootTreeLeafWithMasterDataTests()
+        {
+            var kernel = TestIoContainer.Initialize();
+
+            kernel.Bind(typeof(ITemporalDataDescriptor<>)).To<TemporalDataDescriptor>();
+
+            kernel.Bind<IConvertRegistrations<Root, TargetRoot, ITestConvertIntention>>().To<RootTargetRootConvertRegistrations>();
+
+            kernel.Bind<IConvertRegistrations<Tree, TargetTree, ITestConvertIntention>>().To<TreeTargetTreeConvertRegistrations>();
+            kernel.Bind<IConvertRegistrations<TreeMasterData, TargetTree, ITestConvertIntention>>().To<TreeMasterDataToTreeConvertRegistrations>();
+
+            kernel.Bind<IConvertRegistrations<Leaf, TargetLeaf, ITestConvertIntention>>().To<LeafTargetLeafConvertRegistrations>();
+            kernel.Bind<IConvertRegistrations<LeafMasterData, TargetLeaf, ITestConvertIntention>>().To<LeafMasterDattargetLeafConvertRegistrations>();
+            kernel.Bind<IConvertRegistrations<TemporalLeafMasterData, TargetLeaf, ITestConvertIntention>>().To<TemporalLeafMasterDataTemporalLeafDatconvertRegistrations>();
+
+            kernel.Bind<ICopyRegistrations<ITemporalData>>().To<TemporalDatcopyRegistrations>();
+
+            this.testcandidate = kernel.Get<IConvert<Root, TargetRoot, ITestConvertIntention>>();
+        }
+
+        [Fact]
+        public void StructureFromRootToLeafIsConvertIntoTargetStructure()
+        {
+            // Arrange
+            // Create objects
+            var root = new Root();
+            var tree = new Tree();
+            var treemasterdata = new TreeMasterData();
+            var leaf = new Leaf()
+            {
```

Comment:
Confusing naming. Mix between example names and technical names. Only the leafs give a real hint what it is mean. (Maybe for the next time...)

Reply:
Yeah next time